pkg/utils/kube/resource_ops.go

```diff
@@ -131,6 +124,30 @@ func (k *kubectlResourceOperations) runResourceCommand(ctx context.Context, obj
 	return strings.Join(out, ". "), nil
 }
 
+// rbacReconcile will perform reconciliation for RBAC resources. It will run
+// the following command:
+//
+//	kubectl auth reconcile
+//
+// This is preferred over `kubectl apply`, which cannot tolerate changes in
+// roleRef, which is an immutable field.
+// See: https://github.com/kubernetes/kubernetes/issues/66353
+// `auth reconcile` will delete and recreate the resource if necessary
+func (k *kubectlResourceOperations) rbacReconcile(ctx context.Context, obj *unstructured.Unstructured, fileName string, dryRunStrategy cmdutil.DryRunStrategy) (string, error) {
+	outReconcile, err := func() (string, error) {
+		cleanup, err := k.processKubectlRun("auth")
+		if err != nil {
+			return "", err
+		}
+		defer cleanup()
+		return k.authReconcile(ctx, obj, fileName, dryRunStrategy)
+	}()
```

Comment:
Does this still need to be wrapped in a function? I'm guessing the function existed before to trigger the cleanup call, but this function should behave the same whether or not we wrap the contents in an extra function.

Reply:
I extracted the code in an external function for clarity but didn't change it. Your suggestion makes sense. Changed.